source/common/router/rds_impl.cc

```diff
@@ -119,5 +125,45 @@ void RdsRouteConfigProviderImpl::registerInitTarget(Init::Manager& init_manager)
   init_manager.registerTarget(*this);
 }
 
+HttpRouteManagerImpl::HttpRouteManagerImpl(Runtime::Loader& runtime, Event::Dispatcher& dispatcher,
+                                           Runtime::RandomGenerator& random,
+                                           const LocalInfo::LocalInfo& local_info,
+                                           ThreadLocal::SlotAllocator& tls)
+    : runtime_(runtime), dispatcher_(dispatcher), random_(random), local_info_(local_info),
+      tls_(tls) {}
+
+std::vector<Router::RouteConfigProviderSharedPtr> HttpRouteManagerImpl::routeConfigProviders() {
+  std::vector<Router::RouteConfigProviderSharedPtr> ret;
+  ret.reserve(route_config_providers_.size());
+  for (const auto& element : route_config_providers_) {
+    ret.push_back(element.second.lock());
```

Comment:
Well, yeah, threading isn't relevant, `lock` is a misnomer. It's a way to turn weak into a real shared pointer. Assuming this is weak for a reason, it seems it should be possible for the `lock` to fail... or did I miss something here?